src/coreclr/tools/cdac-build-tool/README.md

```diff
@@ -0,0 +1,133 @@
+
+```mermaid
+flowchart TB
+  headers("runtime headers")
+  data_src("data-descriptor.c")
+  compile_data["clang"]
+  data_obj("data-descriptor.o")
+  contracts("contracts.txt")
+  globals("contract-aux-data.c")
+  build[["cdac-build-tool"]]
+  descriptor_src("contract-descriptor.c")
+  vm("runtime sources")
+  compile_runtime["clang"]
+  runtime_lib(["libcoreclr.so"])
+
+  headers -.-> data_src
+  headers -.-> globals
+  headers -.-> vm
+  data_src --> compile_data --> data_obj --> build
+  contracts ----> build
+  build --> descriptor_src
+  descriptor_src --> compile_runtime
+  globals -----> compile_runtime
+  vm ----> compile_runtime --> runtime_lib
+```
+
+
+## Specifying data descriptors
+
+The sample in the `sample` dir uses the following syntax (see [sample/sample.data.h](sample/sample.data.h)) to specify the data descriptor:
+
+```c
+CDAC_BASELINE("empty")
+CDAC_TYPES_BEGIN()
+
+CDAC_TYPE_BEGIN(ManagedThread)
+CDAC_TYPE_INDETERMINATE(ManagedThread)
+CDAC_TYPE_FIELD(ManagedThread, GCHandle, GCHandle, offsetof(ManagedThread,m_gcHandle))
+CDAC_TYPE_FIELD(ManagedThread, pointer, Next, offsetof(ManagedThread,m_next))
+CDAC_TYPE_END(ManagedThread)
+
+CDAC_TYPE_BEGIN(GCHandle)
+CDAC_TYPE_SIZE(sizeof(intptr_t))
+CDAC_TYPE_END(GCHandle)
+
+CDAC_TYPES_END()
+
+CDAC_GLOBALS_BEGIN()
+// FIXME: wasm32 doesn't like uint64_t cast from uintptr_t at compile-time
+// The right thing to do is to not do pointers using this mechanism since they need to go into
+// auxdata anyway.
+CDAC_GLOBAL_POINTER(ManagedThreadStore, &g_managedThreadStore)
+#if FEATURE_EH_FUNCLETS
+CDAC_GLOBAL(FeatureEHFunclets, uint8, 1)
+#else
+CDAC_GLOBAL(FeatureEHFunclets, uint8, 0)
+#endif
+CDAC_GLOBAL(SomeMagicNumber, uint32, 42)
+CDAC_GLOBALS_END()
+```
+
+**TODO**: finish documenting this
+
+The file is included multiple times with the macros variously defined in order to generate the
+data descriptor blob.
+
+## Implementation Details
+
+See [data-descriptor-blob.md](./data-descriptor-blob.md)
+
+## Workflow
+
+### Porting and extending the data blob scraper
+
+When porting to a new architecture, or extending the blob contents, it is recommended to
+first work withh the sample blob, rather than the full CoreCLR descriptor.
+
+For example, if your target platform has a clang toolchain, something like this will provide a suitable
+input for `cdac-build-tool`:
+
+```console
+$ clang -target wasm32-unknown-unknown -c -o /tmp/sample.o ./sample/sample.blob.c
+```
+
+If you are modifying the preprocessor macros, using `-E` to emit the preprocessed output is helpful as well.
+
+```console
+$ clang -target x86_64-unknown-linux-gnu -E -o /tmp/sample.i ./sample/sample.blob.c
+```
+
+Running the `cdac-build-tool` with the `-v` verbose option will show progress
+
+```console
+$ ../../../../dotnet.sh run --project cdac-build-tool.csproj -- compose -v -o /tmp/contract.c /tmp/sample.o
```

Comment:
I would go in the other direction. Assume this is at the room and point to the `csproj`. It is must easier to look for specific paths to update as opposed to arbitrary parent directory chains.